extensions/amp-story-shopping/0.1/amp-story-shopping-attachment.js

```diff
@@ -295,6 +322,20 @@ export class AmpStoryShoppingAttachment extends AMP.BaseElement {
     });
   }
 
+  /**
+   * onclick event that fires when buy now is clicked, has conversion tracking.
+   * @param {?string} conversionURL
+   * @return {boolean}
+   * @private
+   */
+  onClickBuyNow_(conversionURL) {
+    this.analyticsService_.triggerEvent(
```

Comment:
You can check the interactive components for an example on the variables:
```js

this.variableService_.onVariableUpdate(
      AnalyticsVariable.STORY_INTERACTIVE_TYPE,
      this.interactiveType_
);
```

with sth similar to that, you can add the variables needed before calling `triggerEvent`.